### PR TITLE
[JENKINS-73885] Remove inline `onclick` handler from script removal form

### PR DESCRIPTION
**Pull request by yaroslavafenkin**
https://issues.jenkins.io/browse/JENKINS-73885

### Testing done
I went to the Scriptler management screen as an admin user, added new script, then removed it. CSP plugin is installed in my Jenkins instance, and it registered multiple CSP violations before the change. After the change the list is clear.

https://www.loom.com/share/690a3867e06a42e7ba160f31701eb38c?sid=2a371698-8b64-4aa2-ad62-83c1d0e9e2ad

https://www.loom.com/share/0336da4251ad4419986f6559771494b7?sid=761d9c6b-69a5-490d-b401-d057646e63cb

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] ~Link to relevant pull requests, esp. upstream and downstream changes~
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/resources/org/jenkinsci/plugins/scriptler/ScriptlerManagement/confirm-remove.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const removeScriptButtons = document.querySelectorAll(".scriptler-remove-script-form");
+    removeScriptButtons.forEach(button => button.addEventListener("click", (e) => {
+        const name = e.currentTarget.getAttribute('data-name');
+        if (!confirm("Sure you want to delete [" + name + "]?")) {
+            e.preventDefault();
+        }
+    }));
+});
```
Comment (yaroslavafenkin) [inline src/main/resources/org/jenkinsci/plugins/scriptler/ScriptlerManagement/confirm-remove.js:9]: Content of the script before the change for easier visual comparison:

```js
function confirmDelete(element) {
  var name = element.getAttribute('data-name');
  if (confirm("Sure you want to delete ["+name+"]?")) {
    return true;
  }else{
    return false;
  }
}
```

File: src/main/resources/org/jenkinsci/plugins/scriptler/ScriptlerManagement/confirm-remove.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const removeScriptButtons = document.querySelectorAll(".scriptler-remove-script-form");
+    removeScriptButtons.forEach(button => button.addEventListener("click", (e) => {
+        const name = e.currentTarget.getAttribute('data-name');
+        if (!confirm("Sure you want to delete [" + name + "]?")) {
+            e.preventDefault();
```
Comment (yaroslavafenkin) [inline src/main/resources/org/jenkinsci/plugins/scriptler/ScriptlerManagement/confirm-remove.js:6]: We don't have to do anything if `confirm` is true, just let the form get submitted. If it was cancelled there we prevent the form submission. So I thought reverting the `if` makes sense here.

File: src/main/resources/org/jenkinsci/plugins/scriptler/ScriptlerManagement/confirm-remove.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const removeScriptButtons = document.querySelectorAll(".scriptler-remove-script-form");
+    removeScriptButtons.forEach(button => button.addEventListener("click", (e) => {
```
Comment (basil) [inline src/main/resources/org/jenkinsci/plugins/scriptler/ScriptlerManagement/confirm-remove.js:3]: I was reviewing other similar changes and found that some of them use `Behaviour.specify` in cases like this (e.g., jenkinsci/matrix-project-plugin#213). I'm not sure when `Behaviour.specify` should be used and when it shouldn't. Should we use it here to be consistent with Jenkins core and that other PR? Please feel free to disregard this comment if I am wrong about this.

Comment (basil) [reply src/main/resources/org/jenkinsci/plugins/scriptler/ScriptlerManagement/confirm-remove.js:3]: I found an explanation here: https://www.jenkins.io/doc/developer/security/csp/#inline-event-handlers

**Review by basil — APPROVED** [review]
Thanks for the PR!

**Comment by yaroslavafenkin** [comment]
ATH passed: https://github.com/jenkinsci/acceptance-test-harness/pull/1778

**Pull request merged**